scripts/runnable_recipe_ray_vllm_weight_sync.py

```diff
@@ -368,6 +369,447 @@ def run(self):
+            replay_buffer=replay_buffer,
+            total_frames=total_steps,
+            local_weight_updater=local_weight_updater,
+            remote_weight_updater=remote_weight_updater,
+            reset_at_each_iter=reset_at_each_iter,
+            use_buffers=False,
+            # FIXME: write out what was wrong here...had to set trust_policy to prevent error
+            trust_policy=True,
+        )
+        
+        print("done init LLMCollector")
+    
+    @property
+    def remote_weight_updater(self) -> RemoteWeightUpdaterBase:
+        return self._remote_weight_updater
+    
+    @remote_weight_updater.setter
+    def remote_weight_updater(self, value: RemoteWeightUpdaterBase | None):
+        # if value is not None:
+        #     if isinstance(value, ray.actor.ActorHandle):
+        #         value.register_collector.remote(self)
+        #     else:
+        #         value.register_collector(self)
+        #         if value.collector is not self:
+        #             raise RuntimeError("Failed to register collector.")
+        self._remote_weight_updater = value
+
+    def _postprocess_for_queue(self, data):
+        '''
+        This is a helper that should be deleted once the TensorClass stuff has been figured out.
+        '''
+        data = data.squeeze()
+        query_responses = torch.cat([data["tokens"], data["tokens_response"]], dim=-1)
+        prompt_tokens = data["tokens"]
+        response_tokens = data["tokens_response"]
+        logprobs = data["log_probs"]
+        query_response_padding_masks = torch.ne(
+            query_responses, self._tokenizer.pad_id
+        )
+        answers = data["answers"]
+        if hasattr(self.llm.llm_engine.model_executor.driver_worker.worker, "policy_version"):
+            policy_version = self.llm.llm_engine.model_executor.driver_worker.worker.policy_version.item()
+        else:
+            policy_version = 0
+
+        response_padding_masks = torch.eq(response_tokens, self._tokenizer.pad_id)
+        seq_lens = training.get_unmasked_sequence_lengths(response_padding_masks)
+        del response_padding_masks
+
+        postprocessed_results = (
+            query_responses,
+            response_tokens,
+            logprobs,
+            query_response_padding_masks,
+            seq_lens,
+            answers,
+            policy_version
+        )
+        return postprocessed_results
+
+
+    def run(self):
+        for i in range(63):
+            self.rollout()
+            print(f"{self.worker_id} about to update weights, {self.remote_weight_updater}")
+            if i > 1:
+                self.remote_weight_updater.update_weights.remote(weights=None, worker_ids=self.worker_id)
+                self.local_weight_updater()
+
+
+    def rollout(self) -> TensorDictBase:
+        if self.reset_at_each_iter or self._shuttle is None:
+            data = self.env.reset()
+        else:
+            data = self._shuttle
+
+        tensordicts = []
```

Comment:
We should give a more descriptive name for these beyond their datatype: you wouldn't call a list of integers `ints` :D 

`trajectories`?

Reply:
looks good, or `rollouts`